blscosi/bdnproto/protocol.go

```diff
@@ -0,0 +1,82 @@
+// Package bdnproto implements the Boneh-Drijvers-Neven signature scheme
+// to protect the aggregates against rogue public-key attacks.
+// This is a modified version of blscosi/protocol which is now deprecated.
+package bdnproto
+
+import (
+	"go.dedis.ch/cothority/v3/blscosi/protocol"
+	"go.dedis.ch/kyber/v3/pairing"
+	"go.dedis.ch/kyber/v3/sign"
+	"go.dedis.ch/kyber/v3/sign/bdn"
+	"go.dedis.ch/onet/v3"
+)
+
+// BdnProtocolName is the name of the main protocol for the modified BLS signature scheme
+const BdnProtocolName = "bdnCoSiProto"
+
+// BdnSubProtocolName is the name of the subprotocol for the modified BLS signature scheme
+const BdnSubProtocolName = "bdnSubCosiProto"
+
+func init() {
+	GlobalRegisterBdnProtocols()
+}
+
+// GlobalRegisterBdnProtocols registers both protocol to the global register
+func GlobalRegisterBdnProtocols() {
+	onet.GlobalProtocolRegister(BdnProtocolName, NewBdnProtocol)
+	onet.GlobalProtocolRegister(BdnSubProtocolName, NewSubBdnProtocol)
+}
+
+// NewBdnProtocol is used to register the protocol with an always-true verification
```

Comment:
I'm unclear what this API is for; it seems like we know we'll only be using this in the context where we use a real verifier, and anyway a caller who wants "always verify" will be able to easily provider the no-op vf.

Reply:
It seems that this is the historical way to provide the protocol with an empty verification function, mainly for the tests. I think it's safe to say this is only used for testing so the registration could be effective only for the tests.

Reply:
This is true for all CoSi-related packages